test/unittests/t_catalog_mgr.cc

```diff
@@ -0,0 +1,174 @@
+                          reinterpret_cast<const unsigned char*>(hashes[3]),
+                          size, suffix);
+    new_catalog->AddFile(hash, file_size, "/dir/dir/dir", "file4");
+    // we haven't mounted the second catalog yet!
+    ASSERT_EQ(1, catalog_mgr_->GetNumCatalogs());
+  }
+
+ protected:
+  const static char *hashes[];
+  MockCatalogManager *catalog_mgr_;
+  perf::Statistics statistics_;
+};
+
+const char *T_CatalogManager::hashes[] = {
+     "b026324c6904b2a9cb4b88d6d61c81d1000000",
+     "26ab0db90d72e28ad0ba1e22ee510510000000",
+     "6d7fce9fee471194aa8b5b6e47267f03000000",
+     "48a24b70a0b376535542b996af517398000000",
+     "1dcca23355272056f04fe8bf20edfce0000000"
+};
+
+
+
+TEST_F(T_CatalogManager, InitialConfiguration) {
+  EXPECT_TRUE(catalog_mgr_->Init());
+  EXPECT_EQ(1, catalog_mgr_->GetNumCatalogs());
+  EXPECT_EQ(1u, catalog_mgr_->GetRevision());
+  EXPECT_FALSE(catalog_mgr_->GetVolatileFlag());
+  EXPECT_EQ(0u, catalog_mgr_->GetTTL());
+}
+
+TEST_F(T_CatalogManager, Statistics) {
+  EXPECT_TRUE(catalog_mgr_->Init());
+  Statistics st = catalog_mgr_->statistics();
+  EXPECT_EQ(0u, st.n_listing->Get());
+  EXPECT_EQ(0u, st.n_lookup_inode->Get());
+  EXPECT_EQ(0u, st.n_lookup_path->Get());
+  EXPECT_EQ(0u, st.n_lookup_path_negative->Get());
+  EXPECT_EQ(0u, st.n_lookup_xattrs->Get());
+  EXPECT_EQ(0u, st.n_nested_listing->Get());
+}
+
+TEST_F(T_CatalogManager, InodeConfiguration) {
+  InodeGenerationAnnotation annotation;
+  catalog_mgr_->SetInodeAnnotation(&annotation);
+  EXPECT_TRUE(catalog_mgr_->Init());
+}
+
+TEST_F(T_CatalogManager, Lookup) {
+  catalog::DirectoryEntry dirent;
+  ASSERT_TRUE(catalog_mgr_->Init());
+  AddTree();
+  EXPECT_TRUE(catalog_mgr_->LookupPath("/dir", kLookupSole, &dirent));
+  EXPECT_TRUE(dirent.IsDirectory());
+  EXPECT_TRUE(catalog_mgr_->LookupPath("/dir/dir", kLookupSole, &dirent));
+  EXPECT_TRUE(dirent.IsDirectory());
+  EXPECT_TRUE(catalog_mgr_->LookupPath("/file1", kLookupSole, &dirent));
+  EXPECT_TRUE(dirent.IsRegular());
+  // the father directory belongs to the catalog, so there is no problem
+  EXPECT_TRUE(catalog_mgr_->LookupPath("/dir/dir/file2", kLookupFull, &dirent));
+  EXPECT_TRUE(dirent.IsRegular());
+  // /dir/dir/dir/file4 belongs to a catalog that is not mounted yet
+  EXPECT_TRUE(catalog_mgr_->LookupPath("/dir/dir/dir/file4", kLookupSole,
+                                       &dirent));
+  // the new catalog should be mounted now
+  EXPECT_EQ(2, catalog_mgr_->GetNumCatalogs());
+
+  // the father directory should also belong to the nested catalo
+  EXPECT_TRUE(catalog_mgr_->LookupPath("/dir/dir/dir/file4", kLookupFull,
+                                       &dirent));
+  // it is not a symplink, so it should crash
+  EXPECT_DEATH(catalog_mgr_->LookupPath("/dir/dir/dir/file4", kLookupRawSymlink,
+                                         &dirent), ".*");
+  EXPECT_EQ(2, catalog_mgr_->GetNumCatalogs());
+}
+
+TEST_F(T_CatalogManager, Listing) {
```

Comment:
Perhaps add a second `Listing` test that tries to list a non-existent directory inside the nested catalog and checks if the catalog was properly loaded.